test/integration-serial/run_serial_test.rb

```diff
@@ -158,4 +144,28 @@ def test_crd_can_fail
     apiextensions_v1beta1_kubeclient.delete_custom_resource_definition("mail.stable.example.io")
     apiextensions_v1beta1_kubeclient.delete_custom_resource_definition("others.stable.example.io")
   end
+
+  def test_crd_pruning
+    assert_deploy_success(deploy_fixtures("crd", subset: %w(mail.yml widgets.yml)))
+    assert_logs_match_all([
+      "Phase 1: Initializing deploy",
+      "Detected non-namespaced resources which will never be pruned:",
+      " - CustomResourceDefinition/mail.stable.example.io",
+      "Phase 3: Deploying all resources",
+      "CustomResourceDefinition/mail.stable.example.io (timeout: 120s)",
+      %r{CustomResourceDefinition/mail.stable.example.io\s+Names accepted}
+    ])
+    assert_deploy_success(deploy_fixtures("crd", subset: %w(mail_cr.yml widgets_cr.yml configmap-data.yml)))
+    # Deploy any other non-priority (predeployable) resource to trigger pruning
+    assert_deploy_success(deploy_fixtures("crd", subset: %w(configmap-data.yml configmap2.yml)))
+
+    refute_logs_match("The following resources were pruned: widget \"my-first-mail\"")
```

Comment:
wouldn't it say `mail "my-first-mail"` if it had been? This brings up that we should add an assertion that actually confirms the thing is still there by looking at the namespace.